src/main/java/org/openstreetmap/atlas/geography/atlas/change/diff/AtlasDiffHelper.java

```diff
@@ -0,0 +1,771 @@
+package org.openstreetmap.atlas.geography.atlas.change.diff;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.PolyLine;
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.bloated.BloatedArea;
+import org.openstreetmap.atlas.geography.atlas.bloated.BloatedEdge;
+import org.openstreetmap.atlas.geography.atlas.bloated.BloatedLine;
+import org.openstreetmap.atlas.geography.atlas.bloated.BloatedNode;
+import org.openstreetmap.atlas.geography.atlas.bloated.BloatedPoint;
+import org.openstreetmap.atlas.geography.atlas.bloated.BloatedRelation;
+import org.openstreetmap.atlas.geography.atlas.change.ChangeType;
+import org.openstreetmap.atlas.geography.atlas.change.FeatureChange;
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.Line;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.geography.atlas.items.Point;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.geography.atlas.items.RelationMember;
+import org.openstreetmap.atlas.geography.atlas.items.RelationMemberList;
+import org.openstreetmap.atlas.geography.matching.PolyLineRoute;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A helper class for {@link AtlasDiff}. Contains lots of static utilities.
+ *
+ * @author lcram
+ */
+public final class AtlasDiffHelper
+{
+    private static final Logger logger = LoggerFactory.getLogger(AtlasDiffHelper.class);
+
+    public static boolean edgeHasGeometryMatchAmong(final Edge edge,
+            final Iterable<Edge> otherEdges, final boolean useGeometryMatching)
+    {
+        if (useGeometryMatching)
+        {
+            final PolyLine source = edge.asPolyLine();
+            final List<PolyLine> candidates = Iterables.stream(otherEdges).map(Edge::asPolyLine)
+                    .collectToList();
+            final Optional<PolyLineRoute> match = source.costDistanceToOneWay(candidates)
+                    .match(Distance.ZERO);
+            if (match.isPresent() && match.get().getCost().isLessThanOrEqualTo(Distance.ZERO))
+            {
+                // The edge was probably split by way sectioning without changing itself.
+                logger.trace("Edge {} from {} has no equal member but found a match with no cost.",
+                        edge, edge.getAtlas().getName());
+                return true;
```

Comment:
If I remember correctly, this one was to ignore the ways that are way-sectioned differently but still have the same geometry overall. In the Change case, those would still need to have FeatureChange objects after all. Is that something that is covered after?

Reply:
Let's definitely discuss this further, I am not sure I am 100 percent clear.